compiler/rustc_mir/src/borrow_check/diagnostics/mutability_errors.rs

```diff
@@ -241,6 +244,39 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
                     format!("mut {}", self.local_names[local].unwrap()),
                     Applicability::MachineApplicable,
                 );
+                let tcx = self.infcx.tcx;
+                // point to span of upvar making closure call require mutable borrow
+                if let ty::Closure(id, _) = the_place_err.ty(self.body, tcx).ty.kind() {
+                    let id = id.expect_local();
+                    let tables = tcx.typeck(id);
+                    let hir_id = tcx.hir().local_def_id_to_hir_id(id);
+                    let (span, place) = &tables.closure_kind_origins()[hir_id];
+                    let reason = if let PlaceBase::Upvar(upvar_id) = place.base {
+                        let upvar = ty::place_to_string_for_capture(tcx, place);
+                        match tables.upvar_capture(upvar_id) {
+                            ty::UpvarCapture::ByRef(ty::UpvarBorrow {
+                                kind: ty::BorrowKind::MutBorrow,
+                                ..
+                            }) => {
+                                format!("mutable borrow of `{}`", upvar)
+                            }
+                            ty::UpvarCapture::ByValue(_) => {
+                                format!("possible mutation of `{}`", upvar)
```

Comment:
Why 'possible'? Can there be false positives here, or are you thinking of something like:

```rust
let cb = || {
	if false {
        capture = new_val;
    }
}
```

If it's the latter, I would get rid of the word 'possible', since this can also happen for a mutable borrow (where we don't say 'possible').

Reply:
I was thinking of something like `issue-80313-mutable-borrow-in-move-closure.rs`, where a mutable borrow is taken of the owned upvar, but no mutation occurs. In the case of a normal closure, even in your example, where the mutation is not reachable, I think the mutable borrow is always done.